src/pandapipes/pf/pipeflow_setup.py

```diff
@@ -521,64 +519,38 @@ def identify_active_nodes_branches(net, hydraulic=True):
+                                                                     branches_connected, nodes_connected,
                                                                      mode="heat_transfer")
-        else:
-            # if no full connectivity check is performed, all nodes that are not connected to the
-            # rest of the network wrt. flow can be identified by a more performant sum_by_group_call
-            # check for branches that are not traversed (for temperature calculation, this means
-            # that they are "out of service")
-
-            if get_net_option(net, "transient"):
-                branches_connected = np.copy(get_lookup(net, "branch", "active_hydraulics"))
-                nodes_connected = np.copy(get_lookup(net, "node", "active_hydraulics"))
-            else:
-                fn = branch_pit[:, FROM_NODE].astype(np.int32)
-                tn = branch_pit[:, TO_NODE].astype(np.int32)
-                branches_connected = get_lookup(
-                    net, "branch", "active_hydraulics"
-                ) & branches_connected_flow(branch_pit)
-                fn_tn, flow = _sum_by_group(
-                    get_net_option(net, "use_numba"),
-                    np.concatenate([fn, tn]),
-                    np.concatenate([branches_connected, branches_connected]).astype(
-                        np.int32
-                    ),
-                )
-                nodes_connected = np.copy(get_lookup(net, "node", "active_hydraulics"))
-                # set nodes oos that are not connected to any branches with flow > 0 (0.1 is arbitrary
-                # here, any value between 0 and 1 should work, excluding 0 and 1)
-                nodes_connected[fn_tn] = nodes_connected[fn_tn] & (flow > 0.1)
+        fn = branch_pit[:, FROM_NODE].astype(np.int32)
+        tn = branch_pit[:, TO_NODE].astype(np.int32)
+        branches_flow = branches_connected & branches_connected_flow(branch_pit)
+        nodes_flow = np.isin(np.arange(len(nodes_connected)), np.concatenate([fn[branches_flow], tn[branches_flow]])
+        ) & nodes_connected
 
         if get_net_option(net, "transient"):
-            fn = branch_pit[:, FROM_NODE].astype(np.int32)
-            tn = branch_pit[:, TO_NODE].astype(np.int32)
-            branches_zero = branches_zero_flow(branch_pit)
-            connected_branches_not_zero = branches_connected & ~branches_zero
-            nodes_zero = ~np.isin(
-                np.arange(len(nodes_connected)),
-                np.concatenate([fn[connected_branches_not_zero], tn[connected_branches_not_zero]])
-            ) & nodes_connected
-            net["_lookups"]["node_zero_flow"] = nodes_zero
-            net["_lookups"]["branch_zero_flow"] = branches_zero
+            net["_lookups"]["node_zero_flow"] = ~nodes_flow
+            net["_lookups"]["branch_zero_flow"] = ~branches_flow
+        else:
+            branches_connected = branches_flow
+            nodes_connected = nodes_flow
+
```

Comment:
This approach does indeed look much cleaner. But for the zero flow branches / nodes we still need to consider only those active, i.e.

`net["_lookups"]["node_zero_flow"] = ~nodes_flow & nodes_connected`

Reply:
Not required as already done in line 538/539:
```
        nodes_flow = np.isin(np.arange(len(nodes_connected)), np.concatenate([fn[branches_flow], tn[branches_flow]])
        ) & nodes_connected
```